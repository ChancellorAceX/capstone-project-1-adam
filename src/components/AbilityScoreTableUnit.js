import React from 'react';

export default function AbilityScoreTableUnit(props) {
  return (
    <tr>
      <th>{props.unit[0]}</th>
      <th>{props.unit[1]}</th>
      <th>{props.unit[2]}</th>
    </tr>
  )
}