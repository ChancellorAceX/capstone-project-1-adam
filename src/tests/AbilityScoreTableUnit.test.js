import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AbilityScoreTableUnit from '../components/AbilityScoreTableUnit';

describe('AbilityScoreTableUnit component', () => {
  const score = ['Strength', 'str', 'strmod'];
  const monster = { str: 17, strmod: 3 };
  it('renders the complete table unit', () => {
    const wrapper = shallow(<AbilityScoreTableUnit
      unit={[score[0], monster[score[1]], (monster[score[1]] > 12 ? '+' : '') + monster[score[2]]]}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});