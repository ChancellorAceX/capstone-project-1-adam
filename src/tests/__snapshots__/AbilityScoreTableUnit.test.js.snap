// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbilityScoreTableUnit component renders the complete table unit 1`] = `
<tr>
  <th>
    Strength
  </th>
  <th>
    17
  </th>
  <th>
    +3
  </th>
</tr>
`;
